src/main/java/kr/hhplus/be/server/application/bestseller/BestSellerScheduler.java

```diff
@@ -1,73 +1,85 @@
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
+import java.util.Set;
+import java.util.stream.IntStream;
 
 @Slf4j
 @RequiredArgsConstructor
 @Component
 public class BestSellerScheduler {
 
-    private final OrderService orderService;
     private final ProductService productService;
-    private final BestSellerService bestSellerService;
-    private final ApplicationContext applicationContext;
-
-    @Scheduled(cron = "0 0 * * * *")
-    @Transactional
-    public void calculateHourlyBestSellers() {
-        LocalDateTime now = LocalDateTime.now();
-        LocalDateTime oneHourAgo = now.minusHours(1);
-
-        List<Order> orders = orderService.findPaidOrdersBetween(oneHourAgo, now);
-
-        Map<Long, Long> productIdTotalSales = orders.stream()
-                .flatMap(order -> order.getOrderProducts().stream())
-                .collect(Collectors.groupingBy(
-                        OrderProduct::getProductId,
-                        Collectors.summingLong(OrderProduct::getQuantity)
-                ));
-
-        for (Map.Entry<Long, Long> entry : productIdTotalSales.entrySet()) {
-            Long productId = entry.getKey();
-            Long sales = entry.getValue();
-
-            Product product = productService.getProductWithLock(productId);
-            BestSeller bestSeller = BestSeller.of(product, sales);
-            bestSellerService.save(bestSeller);
+    private final SalesRankingRepository salesRankingRepository;
+
+    private static final String DAILY_SALES_PREFIX = "sales:daily:";
+    private static final String LATEST_3DAYS_SALES_PREFIX = "sales:3days:";
+
+    @CachePut(value = CacheNames.DAY3_BEST_SELLERS, key = "'best'", cacheManager = "redisCacheManager")
+    @Scheduled(cron = "0 0/10 * * * *")
+    public BestSellerDto getLatest3DaysTop5() {
+        LocalDate today = LocalDate.now();
+        String latest3daysSalesKey = LATEST_3DAYS_SALES_PREFIX + today.format(DateTimeFormatter.BASIC_ISO_DATE);
+
+        List<String> keys = IntStream.rangeClosed(0, 2)
+                .mapToObj(i -> DAILY_SALES_PREFIX + today.minusDays(i).format(DateTimeFormatter.BASIC_ISO_DATE))
+                .toList();
+
+        Long resultCount = salesRankingRepository.aggregateSales(keys, latest3daysSalesKey);
+        salesRankingRepository.setSalesKeyTtl(latest3daysSalesKey, Duration.ofDays(1));
+
+        log.info("[BestSellerScheduler] 최근 3일간 판매 상품 집계 완료. 총 상품 수 {}", resultCount);
+
+        Set<ZSetOperations.TypedTuple<String>> top5 = salesRankingRepository.getTopProducts(latest3daysSalesKey, 5);
+
+        if (top5 == null || top5.isEmpty()) {
+            log.info("[BestSellerScheduler] 최근 3일간 인기 상품 집계 결과 없음");
+            return BestSellerDto.empty();
         }
 
-        log.info("[BestSellerScheduler] 1시간 동안 주문된 상품 통계 집계 완료");
-    }
+        List<Long> top5Ids = top5.stream()
+                .map(t -> extractProductId(t.getValue()))
+                .toList();
```

Comment:
```suggestion
        List<Long> top5Ids = top5.stream()
                .map(this:extractProductId)
                .toList();
```

위와 같이 바꾸고 아래처럼 메서드 시그니처를 바꾸면 가독성이 더 좋을 것 같아요! 

```
private Long extractProductId(TypedTuple<String> value) {
        if (value == null || !value.startsWith("product:")) {
            throw new ApiException(ErrorCode.INTERNAL_SERVER_ERROR);
        }

        return Long.valueOf(value.substring("product:".length()));
    }
```